test/mocks/stats/mocks.h

```diff
@@ -27,10 +27,13 @@ class MockCounter : public Counter {
   MockCounter();
   ~MockCounter();
 
+  // Note: cannot be mocked because it is accessed as a Property in a gmock EXPECT_CALL. This
```

Comment:
FWIW I still don't understand what this means and why it happens (I realize this was copied).

Reply:
I don't really understand it either. I think it was introduced in #1803. @mrice32, can you supply context here?